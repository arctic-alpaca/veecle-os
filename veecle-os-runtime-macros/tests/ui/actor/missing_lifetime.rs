#[derive(Debug, PartialEq, Clone, Default, veecle_os_runtime::Storable)]
pub struct Sensor(pub u8);

#[veecle_os_runtime_macros::actor]
async fn macro_test_actor(_reader: veecle_os_runtime::Reader) -> std::convert::Infallible {
    unreachable!("We only care about the code compiling.");
}

#[veecle_os_runtime_macros::actor]
async fn macro_test_actor2(_reader: veecle_os_runtime::Reader) -> std::convert::Infallible {
    unreachable!("We only care about the code compiling.");
}

#[veecle_os_runtime_macros::actor]
async fn macro_test_actor3(_reader: veecle_os_runtime::Reader<Sensor>) -> std::convert::Infallible {
    unreachable!("We only care about the code compiling.");
}

#[veecle_os_runtime_macros::actor]
async fn macro_test_actor4(_reader: veecle_os_runtime::Reader<'b, Sensor>) -> std::convert::Infallible {
    unreachable!("We only care about the code compiling.");
}

#[veecle_os_runtime_macros::actor]
async fn macro_test_actor5(_reader: veecle_os_runtime::Reader) -> std::convert::Infallible {
    unreachable!("We only care about the code compiling.");
}

fn main() {}
