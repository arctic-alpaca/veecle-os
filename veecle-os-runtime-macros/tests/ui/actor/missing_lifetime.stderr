error: lifetimes on actor arguments must be anonymous or static
  --> tests/ui/actor/missing_lifetime.rs:20:63
   |
20 | async fn macro_test_actor4(_reader: veecle_os_runtime::Reader<'b, Sensor>) -> std::convert::Infallible {
   |                                                               ^^

error[E0106]: missing lifetime specifier
 --> tests/ui/actor/missing_lifetime.rs:5:55
  |
5 | async fn macro_test_actor(_reader: veecle_os_runtime::Reader) -> std::convert::Infallible {
  |                                                       ^^^^^^ expected named lifetime parameter
  |
help: consider using the `'__dont_use_internal_actor_macro_lifetime` lifetime
  |
5 | async fn macro_test_actor(_reader: veecle_os_runtime::Reader<'__dont_use_internal_actor_macro_lifetime>) -> std::convert::Infallible {
  |                                                             +++++++++++++++++++++++++++++++++++++++++++

error[E0726]: implicit elided lifetime not allowed here
 --> tests/ui/actor/missing_lifetime.rs:5:36
  |
5 | async fn macro_test_actor(_reader: veecle_os_runtime::Reader) -> std::convert::Infallible {
  |                                    ^^^^^^^^^^^^^^^^^^^^^^^^^ expected lifetime parameter
  |
help: indicate the anonymous lifetime
  |
5 | async fn macro_test_actor(_reader: veecle_os_runtime::Reader<'_>) -> std::convert::Infallible {
  |                                                             ++++

error[E0106]: missing lifetime specifier
  --> tests/ui/actor/missing_lifetime.rs:10:56
   |
10 | async fn macro_test_actor2(_reader: veecle_os_runtime::Reader) -> std::convert::Infallible {
   |                                                        ^^^^^^ expected named lifetime parameter
   |
help: consider using the `'__dont_use_internal_actor_macro_lifetime` lifetime
   |
10 | async fn macro_test_actor2(_reader: veecle_os_runtime::Reader<'__dont_use_internal_actor_macro_lifetime>) -> std::convert::Infallible {
   |                                                              +++++++++++++++++++++++++++++++++++++++++++

error[E0726]: implicit elided lifetime not allowed here
  --> tests/ui/actor/missing_lifetime.rs:10:37
   |
10 | async fn macro_test_actor2(_reader: veecle_os_runtime::Reader) -> std::convert::Infallible {
   |                                     ^^^^^^^^^^^^^^^^^^^^^^^^^ expected lifetime parameter
   |
help: indicate the anonymous lifetime
   |
10 | async fn macro_test_actor2(_reader: veecle_os_runtime::Reader<'_>) -> std::convert::Infallible {
   |                                                              ++++

error[E0106]: missing lifetime specifier
  --> tests/ui/actor/missing_lifetime.rs:15:62
   |
15 | async fn macro_test_actor3(_reader: veecle_os_runtime::Reader<Sensor>) -> std::convert::Infallible {
   |                                                              ^ expected named lifetime parameter
   |
help: consider using the `'__dont_use_internal_actor_macro_lifetime` lifetime
   |
15 | async fn macro_test_actor3(_reader: veecle_os_runtime::Reader<'__dont_use_internal_actor_macro_lifetime, Sensor>) -> std::convert::Infallible {
   |                                                               ++++++++++++++++++++++++++++++++++++++++++

error[E0726]: implicit elided lifetime not allowed here
  --> tests/ui/actor/missing_lifetime.rs:15:37
   |
15 | async fn macro_test_actor3(_reader: veecle_os_runtime::Reader<Sensor>) -> std::convert::Infallible {
   |                                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected lifetime parameter
   |
help: indicate the anonymous lifetime
   |
15 | async fn macro_test_actor3(_reader: veecle_os_runtime::Reader<'_, Sensor>) -> std::convert::Infallible {
   |                                                               +++

error[E0106]: missing lifetime specifier
  --> tests/ui/actor/missing_lifetime.rs:25:56
   |
25 | async fn macro_test_actor5(_reader: veecle_os_runtime::Reader) -> std::convert::Infallible {
   |                                                        ^^^^^^ expected named lifetime parameter
   |
help: consider using the `'__dont_use_internal_actor_macro_lifetime` lifetime
   |
25 | async fn macro_test_actor5(_reader: veecle_os_runtime::Reader<'__dont_use_internal_actor_macro_lifetime>) -> std::convert::Infallible {
   |                                                              +++++++++++++++++++++++++++++++++++++++++++

error[E0726]: implicit elided lifetime not allowed here
  --> tests/ui/actor/missing_lifetime.rs:25:37
   |
25 | async fn macro_test_actor5(_reader: veecle_os_runtime::Reader) -> std::convert::Infallible {
   |                                     ^^^^^^^^^^^^^^^^^^^^^^^^^ expected lifetime parameter
   |
help: indicate the anonymous lifetime
   |
25 | async fn macro_test_actor5(_reader: veecle_os_runtime::Reader<'_>) -> std::convert::Infallible {
   |                                                              ++++

error[E0107]: missing generics for struct `Reader`
 --> tests/ui/actor/missing_lifetime.rs:5:55
  |
5 | async fn macro_test_actor(_reader: veecle_os_runtime::Reader) -> std::convert::Infallible {
  |                                                       ^^^^^^ expected 1 generic argument
  |
note: struct defined here, with 1 generic parameter: `T`
 --> $WORKSPACE/veecle-os-runtime/src/datastore/reader.rs
  |
  | pub struct Reader<'a, T>
  |            ^^^^^^     -
help: add missing generic argument
  |
5 | async fn macro_test_actor(_reader: veecle_os_runtime::Reader<T>) -> std::convert::Infallible {
  |                                                             +++

error[E0107]: missing generics for struct `Reader`
  --> tests/ui/actor/missing_lifetime.rs:10:56
   |
10 | async fn macro_test_actor2(_reader: veecle_os_runtime::Reader) -> std::convert::Infallible {
   |                                                        ^^^^^^ expected 1 generic argument
   |
note: struct defined here, with 1 generic parameter: `T`
  --> $WORKSPACE/veecle-os-runtime/src/datastore/reader.rs
   |
   | pub struct Reader<'a, T>
   |            ^^^^^^     -
help: add missing generic argument
   |
10 | async fn macro_test_actor2(_reader: veecle_os_runtime::Reader<T>) -> std::convert::Infallible {
   |                                                              +++

error[E0107]: missing generics for struct `Reader`
  --> tests/ui/actor/missing_lifetime.rs:25:56
   |
25 | async fn macro_test_actor5(_reader: veecle_os_runtime::Reader) -> std::convert::Infallible {
   |                                                        ^^^^^^ expected 1 generic argument
   |
note: struct defined here, with 1 generic parameter: `T`
  --> $WORKSPACE/veecle-os-runtime/src/datastore/reader.rs
   |
   | pub struct Reader<'a, T>
   |            ^^^^^^     -
help: add missing generic argument
   |
25 | async fn macro_test_actor5(_reader: veecle_os_runtime::Reader<T>) -> std::convert::Infallible {
   |                                                              +++
